packages/@aws-cdk/aws-msk/lib/cluster.ts

```diff
@@ -1,34 +1,776 @@
+  public get zookeeperConnectionStringTls(): string {
+    return this._zookeeperConnectionString('ZookeeperConnectStringTls');
+  }
+
+  /**
+   * Get the list of brokers that a client application can use to bootstrap
+   *
+   * Uses a Custom Resource to make an API call to `getBootstrapBrokers` using the Javascript SDK
+   *
+   * @param responseField Field to return from API call. eg. BootstrapBrokerStringSaslScram, BootstrapBrokerString
+   * @returns - A string containing one or more hostname:port pairs.
+   */
+  private _bootstrapBrokers(responseField: string): string {
+    if (!this._clusterBootstrapBrokers) {
+      this._clusterBootstrapBrokers = new cr.AwsCustomResource(this, `BootstrapBrokers${responseField}`, {
+        onUpdate: {
+          service: 'Kafka',
+          action: 'getBootstrapBrokers',
+          parameters: {
+            ClusterArn: this.clusterArn,
+          },
+          physicalResourceId: cr.PhysicalResourceId.of('BootstrapBrokers'),
+        },
+        policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+          resources: [this.clusterArn],
+        }),
+      });
+    }
+    return this._clusterBootstrapBrokers.getResponseField(responseField);
+
+  }
+  /**
+   * Get the list of brokers that a client application can use to bootstrap
+   *
+   * Uses a Custom Resource to make an API call to `getBootstrapBrokers` using the Javascript SDK
+   *
+   * @returns - A string containing one or more hostname:port pairs.
+   */
+  public get bootstrapBrokers(): string {
+    return this._bootstrapBrokers('BootstrapBrokerString');
+  }
+
+  /**
+   * Get the list of brokers that a TLS authenticated client application can use to bootstrap
+   *
+   * Uses a Custom Resource to make an API call to `getBootstrapBrokers` using the Javascript SDK
+   *
+   * @returns - A string containing one or more DNS names (or IP) and TLS port pairs.
+   */
+  public get bootstrapBrokersTls(): string {
+    return this._bootstrapBrokers('BootstrapBrokerStringTls');
+  }
+
+  /**
+   * Get the list of brokers that a SASL/SCRAM authenticated client application can use to bootstrap
+   *
+   * Uses a Custom Resource to make an API call to `getBootstrapBrokers` using the Javascript SDK
+   *
+   * @returns - A string containing one or more dns name (or IP) and SASL SCRAM port pairs.
+   */
+  public get bootstrapBrokersSaslScram(): string {
+    return this._bootstrapBrokers('BootstrapBrokerStringSaslScram');
+  }
+
+  /**
+   * A list of usersnames to register with the cluster. The password will automatically be generated using Secrets
+   * Manager and the { username, password } JSON object stored in Secrets Manager as `AmazonMSK_username`.
+   *
+   * Must be using the SASL/SCRAM authentication mechanism.
+   *
+   * @param usernames - username(s) to register with the cluster
+   */
+  public addUser(...usernames: string[]): void {
+    if (this.saslScramAuthenticationKey) {
+      const MSK_SECRET_PREFIX = 'AmazonMSK_'; // Required
```

Comment:
Maybe add the cluster name? otherwise this will conflict when the same user is added to two different clusters.